src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -98,7 +98,7 @@ public final class InlineConfigParser {
 
     /** A pattern to find the string: "// violation, explanation". */
     private static final Pattern VIOLATION_WITH_EXPLANATION_PATTERN = Pattern
-            .compile(".*//\\s*violation,\\s.+\\s(?:['\"](.*)['\"])?$");
+            .compile("\".*//\\\\s*violation,\\\\s+(?:.*)?$\"");
```

Comment:
I don't know why, but you encase this string in another set of quotes and looks like you double escaped the escapes.

If you used Eclipse, I would say you pasted the old regexp and string inside the string here in order to bring the old PR forward. Newer versions of Eclipse will escape whatever you paste into a string in Java.

Reply:
Fixed